notifyme-sdk-backend/notifyme-sdk-backend-ws/src/test/java/ch/ubique/notifyme/sdk/backend/ws/insertmanager/insertfilters/FakeRequestFilterTest.java

```diff
@@ -0,0 +1,62 @@
+package ch.ubique.notifyme.sdk.backend.ws.insertmanager.insertfilters;
+
+import ch.ubique.notifyme.sdk.backend.model.UserUploadPayloadOuterClass.UploadVenueInfo;
+import com.google.protobuf.ByteString;
+
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FakeRequestFilterTest extends UploadInsertionFilterTest {
+  @Override
+  List<UploadVenueInfo> getValidVenueInfo() {
+    final var venueInfoList = new ArrayList<UploadVenueInfo>();
+    final var venueInfo = getVenueInfo(false);
+    venueInfoList.add(venueInfo);
+    return venueInfoList;
+  }
+
+  @Override
+  List<UploadVenueInfo> getInvalidVenueInfo() {
+    final var venueInfoList = new ArrayList<UploadVenueInfo>();
+    final var venueInfo = getVenueInfo(true);
+    venueInfoList.add(venueInfo);
+    return venueInfoList;
+  }
+
+  @Override
+  UploadInsertionFilter insertionFilter() {
+    return new FakeRequestFilter();
+  }
+
+  private UploadVenueInfo getVenueInfo(boolean fake) {
+    LocalDateTime start = LocalDateTime.now();
+    LocalDateTime end = start.plusMinutes(30);
+    final var crypto = cryptoWrapper.getCryptoUtilV3();
+    final var noncesAndNotificationKey =
+        crypto.getNoncesAndNotificationKey(crypto.createNonce(256));
+    byte[] preid =
+        crypto.cryptoHashSHA256(
+            crypto.concatenate(
+                "CN-PREID".getBytes(StandardCharsets.US_ASCII),
+                "payload".getBytes(StandardCharsets.US_ASCII),
+                noncesAndNotificationKey.noncePreId));
+    byte[] timekey =
+        crypto.cryptoHashSHA256(
+            crypto.concatenate(
+                "CN-TIMEKEY".getBytes(StandardCharsets.US_ASCII),
+                crypto.longToBytes(3600L),
+                crypto.longToBytes(start.toInstant(ZoneOffset.UTC).getEpochSecond()),
+                noncesAndNotificationKey.nonceTimekey));
```

Comment:
it seems that computation of `preid`, `timekey` etc can be moved to the parent test class to clean this up a little bit. Possibly you want to move almost all of this function to the parent test class and add a few parameters if you need them. That'll also make it a little bit easier to read the tests and see the differences.

Reply:
Resolved: See [https://github.com/SwissCovid/swisscovid-cn-backend/commit/fa318a9f0d77c3763605c59942663fecdd5a24d1](https://github.com/SwissCovid/swisscovid-cn-backend/commit/fa318a9f0d77c3763605c59942663fecdd5a24d1)
Note that the entire repository has been moved to [SwissCovid/swisscovid-cn-backend](https://github.com/SwissCovid/swisscovid-cn-backend) and future work will continue there.